app/src/main/java/com/github/sdp/ratemyepfl/model/items/Event.kt

```diff
@@ -0,0 +1,97 @@
+package com.github.sdp.ratemyepfl.model.items
+
+import com.github.sdp.ratemyepfl.database.EventRepository
+import com.github.sdp.ratemyepfl.database.Repository
+import java.time.LocalDateTime
+
+class Event(
+    override val id: String,
+    override val numReviews: Int,
+    override val averageGrade: Double,
+    val numParticipants: Int,
+    val limitParticipants: Int,
+    val lat: Double,
+    val long: Double,
+    val date: LocalDateTime
+) : Reviewable() {
+
+    override fun toString(): String {
+        return id
+    }
+
+    /**
+     * Creates an hash map of the Event, to add it to the DB
+     */
+    fun toHashMap(): HashMap<String, Any?> {
+        return hashMapOf(
+            EventRepository.NUMBER_PARTICIPANTS_FIELD_NAME to numParticipants.toString(),
+            EventRepository.LIMIT_PARTICIPANTS_FIELD_NAME to limitParticipants.toString(),
+            EventRepository.LATITUDE_FIELD_NAME to lat.toString(),
+            EventRepository.LONGITUDE_FIELD_NAME to long.toString(),
+            EventRepository.DATE_FIELD_NAME to date.toString(),
+            Repository.NUM_REVIEWS_FIELD_NAME to numReviews.toString(),
+            Repository.AVERAGE_GRADE_FIELD_NAME to averageGrade.toString()
+        )
+    }
+
+    /**
+     * Builder to create an event step by step
+     * Mandatory fields are:
+     *  - [id]
+     */
+    class Builder : ReviewableBuilder<Event> {

```

Comment:
This is never used right ??

Reply:
Yes for now it is unused but it will be used when fragments for events will be implemented.

Reply:
I see !